tests/bgp/test_bgp_azng_migration.py

```diff
@@ -105,89 +134,379 @@ def test_bgp_azng_migration(duthosts, enum_upstream_dut_hostname):
+                routes_json['totalPrefixCounter'],
+                bgp_nbr_recv_cmd
+            )
 
         for bgp_nbr_adv_cmd in adv_cmd_list:
             res = duthost.shell(duthost.get_vtysh_cmd_for_namespace(bgp_nbr_adv_cmd, peer_device_namespace))
             routes_json = json.loads(res['stdout'])
-            assert routes_json['totalPrefixCounter'] == len(duthosts)
-            assert routes_json['filteredPrefixCounter'] == 0
+            assert routes_json['totalPrefixCounter'] == len(duthosts), (
+                "Mismatch in total advertised route count after AZNG migration rollback. "
+                "Expected totalPrefixCounter to equal the number of DUT hosts ({}), but got {}. "
+                "Command: {}\n"
+            ).format(
+                len(duthosts),
+                routes_json['totalPrefixCounter'],
+                bgp_nbr_adv_cmd
+            )
+
+            assert routes_json['filteredPrefixCounter'] == 0, (
+                "Filtered prefix counter is not zero after AZNG migration rollback. "
+                "Expected filteredPrefixCounter == 0, but got {}. "
+                "- Command: {}\n"
+            ).format(
+                routes_json['filteredPrefixCounter'],
+                bgp_nbr_adv_cmd if 'bgp_nbr_adv_cmd' in locals() else 'N/A'
+            )
 
         rc = duthost.shell('sudo azng_migration -d')
 
-        pytest_assert(not rc['failed'], "AZNG Migration Deny Route-map apply failed")
+        pytest_assert(
+            not rc['failed'],
+            (
+                "AZNG Migration Deny Route-map apply failed. "
+                "Check the command output, DUT logs, and azng_migration script logs.\n"
+                "- Command return code: {}\n"
+                "- Command output: {}\n"
+                "- Hostname: {}\n"
+                "- Platform: {}\n"
+                "- HWSKU: {}\n"
+            ).format(
+                rc.get('rc', 'N/A'),
+                rc.get('stdout', 'N/A'),
+                duthost.hostname,
+                duthost.facts.get("platform"),
+                duthost.facts.get("hwsku")
+            )
+        )
 
         for bgp_nbr_recv_cmd in recv_cmd_list:
             res = duthost.shell(duthost.get_vtysh_cmd_for_namespace(bgp_nbr_recv_cmd, peer_device_namespace))
             routes_json = json.loads(res['stdout'])
-            assert routes_json['totalPrefixCounter'] > 0
-            assert routes_json['filteredPrefixCounter'] == routes_json['totalPrefixCounter']
+            assert routes_json['totalPrefixCounter'] > 0, (
+                "No BGP routes were found (totalPrefixCounter <= 0) in the command output. "
+                "Expected totalPrefixCounter > 0, but got {}. "
+                "- Command: {}\n"
+                "- Output: {}\n"
+            ).format(
+                routes_json['totalPrefixCounter'],
+                bgp_nbr_recv_cmd if 'bgp_nbr_recv_cmd' in locals() else 'N/A',
+                routes_json
```

Comment:
Don't print routes_json out, just printing out routes_json['totalPrefixCounter'] is enough